src/utils.cpp

```diff
@@ -120,7 +122,12 @@ steps_range steps_range::parse(const std::string& string) {
 
         if (splitted.size() == 3) {
             if (splitted[2] != "") {
-                auto stride = std::string2ll(splitted[2]);
+                long long stride = -1;
+                try{
```

Comment:
A small bit of formatting: `try {` (with a space). 

FYI, you can wrap multiple statements in a try/catch block:

```cpp
try {
    var = foo();
    do_stuff(var, baz(45));
} catch (...) {
     // Get here if any operation throws
}
```